src/app/components/SwapSubmitModal/view.jsx

```diff
@@ -76,4 +81,29 @@ const SwapSubmitModal = ({
   </div>
 )
 
-export default SwapSubmitModal
+export default compose(
```

Comment:
I like that you converted this to recompose. Could you combine the index.jsx and view.jsx into a single component? In index.jsx the values calculated inside render can be added using `withProps` and the handlers using `withHandlers`. That'll eliminate some of the prop types and default props you defined below and make it clearer what props should be passed in externally.